### PR TITLE
ref(all) expose cleanup functions

### PR DESCRIPTION
**Pull request by quitrk**

### CODE REVIEW

File: powermonitor/main.js [hunk 1]
```diff
@@ -53,48 +53,57 @@ function systemIdleErrorResult(id, error) {
...
  */
-module.exports = function setupPowerMonitorMain(jitsiMeetWindow) {
+function setupPowerMonitorMain(jitsiMeetWindow) {
     app.whenReady().then(() => {
         _attachEvents(jitsiMeetWindow);
     });
 
-    ipcMain.on(POWER_MONITOR_QUERIES_CHANNEL, (source, { id, data }) => {
-        const { powerMonitor } = electron;
+    ipcMain.on(POWER_MONITOR_QUERIES_CHANNEL, cleanup);
```
Comment (saghul) [inline powermonitor/main.js:101]: Did you intend to call this function cleanup?

File: powermonitor/main.js [hunk 2]
```diff
@@ -53,48 +53,57 @@ function systemIdleErrorResult(id, error) {
...
 }
 
+function cleanup(source, { id, data }) {
+    const { powerMonitor } = electron;
+
+    switch(data.type) {
+        case METHODS.queryIdleState:
+            if (typeof powerMonitor.getSystemIdleState === 'function') { // electron 5+
+                systemIdleResult(id, powerMonitor.getSystemIdleState(data.idleThreshold));
+            } else { // electron 4 or older
```
Comment (saghul) [inline powermonitor/main.js:63]: Since you are moving code around, can you pl drop this branch?

File: powermonitor/main.js [hunk 3]
```diff
@@ -53,48 +53,57 @@ function systemIdleErrorResult(id, error) {
...
-                } else { // electron 4 or older
-                    powerMonitor.querySystemIdleTime(
-                        idleTime => {
-                            systemIdleResult(id, idleTime);
-                        });
-                }
-                break;
-            default: {
-                const error = 'Unknown event type!';
+    jitsiMeetWindow.on('close', cleanup);
```
Comment (saghul) [inline powermonitor/main.js:103]: This doesn't remove the IPC listener...

File: remotecontrol/main.js [hunk 4]
```diff
@@ -1,29 +1,57 @@
...
+        // Clean up ipcMain handlers to avoid leaks.
+        this._jitsiMeetWindow.on('closed', this.cleanup);
+    }
+
+    /**
+     * Cleanup any handlers
+     */
+     cleanup() {
+        ipcMain.removeListener(GET_DISPLAY_EVENT, this._handleGetDisplayEvent);
+        screen.removeListener(DISPLAY_METRICS_CHANGED, this._handleDisplayMetricsChanged);
```
Comment (saghul) [inline remotecontrol/main.js:36]: Could this fail if the app ended before it was ever ready?

**Review by saghul — CHANGES_REQUESTED** [review]

File: powermonitor/main.js (hunk 1 above)
Comment (quitrk) [reply powermonitor/main.js:101]: whoops

File: remotecontrol/main.js (hunk 4 above)
Comment (quitrk) [reply remotecontrol/main.js:36]: i find it unlikely, unless someone calls this function intentionally before that happens, which makes no sense.

**Review by saghul — APPROVED** [review]

**Pull request merged**